flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/BiDirectionalResourceToRequirementMapping.java

```diff
@@ -74,23 +81,26 @@ private static void internalDecrementCount(
                             "Attempting to decrement count of %s->%s, but primary key was unknown.",
                             resourceProfile,
                             secondaryKey);
-                    resourceCounter.decrementCount(secondaryKey, decrement);
-                    return resourceCounter.isEmpty() ? null : resourceCounter;
+                    final ResourceCounter newCounter =
+                            resourceCounter.subtract(secondaryKey, decrement);
+                    return newCounter.isEmpty() ? null : newCounter;
                 });
     }
 
-    public Map<ResourceProfile, Integer> getResourcesFulfilling(ResourceProfile requirement) {
+    public Collection<Map.Entry<ResourceProfile, Integer>> getResourcesFulfilling(
```

Comment:
Instead of returning `Collection<Map.Entry<ResourceProfile, Integer>>` couldn't we also return a `ResourceCounter`? Given that this class is immutable it should not be a problem to hand it out.

Reply:
Yes, I think we can directly return the `ResourceCounter` since it is immutable now.